src/sentry/static/sentry/app/views/inviteMember/inviteMember.jsx

```diff
@@ -0,0 +1,220 @@
+    emails
+      .reduce((prev, cur_email) => {
+        return prev.then(() => this.inviteUser(cur_email));
+      }, Promise.resolve())
+      .then(() => this.redirectToMemberPage())
+      .catch(error => {
+        if (!error.email && !error.role) {
+          Raven.captureMessage('unkown error ', {
+            extra: {error, state: this.state}
+          });
+        }
+
+        this.setState({error, loading: false});
+      });
+  },
+
+  toggleID(id) {
+    let {selectedTeams} = this.state;
+    if (selectedTeams.has(id)) {
+      selectedTeams.delete(id);
+    } else {
+      selectedTeams.add(id);
+    }
+    this.setState({selectedTeams});
+  },
+
+  renderRoleSelect() {
+    let {roleList, selectedRole} = this.state;
+
+    return (
+      <div className="new-invite-team box">
+        <div className="box-header">
+          <h4>{t('Role') + ':'}</h4>
+        </div>
+        <div className="box-content with-padding">
+          <ul className="radio-inputs">
+            {roleList.map(({role, allowed}, i) => {
+              let {desc, name, id} = role;
+              return (
+                <li
+                  className="radio"
+                  key={id}
+                  onClick={() => allowed && this.setState({selectedRole: id})}
+                  style={allowed ? {} : {color: 'grey', cursor: 'default'}}>
+                  <label style={allowed ? {} : {cursor: 'default'}}>
+                    <Radio id={id} value={name} checked={id === selectedRole} readOnly />
+                    {name}
+                    <div className="help-block">{desc}</div>
+                  </label>
+                </li>
+              );
+            })}
+          </ul>
+        </div>
+      </div>
+    );
+  },
+
+  renderTeamSelect() {
+    let org = this.getOrganization();
+    let {teams} = org;
+    let {selectedTeams} = this.state;
+    //no need to select a team when there's only one option
+    if (teams.length < 2) return null;
+    return (
+      <div className="new-invite-team box">
+        <div className="box-header">
+          <h4>{t('Team') + ':'}</h4>
+        </div>
+        <div className="grouping-controls team-choices row box-content with-padding">
+          {teams.map(({slug, name, id}, i) => (
+            <div
+              key={id}
+              onClick={e => {
+                e.preventDefault();
+                this.toggleID(id);
+              }}
+              className="col-md-3">
+              <label className="checkbox">
+                <Checkbox id={id} value={name} checked={selectedTeams.has(id)} />
+                {name}
+                <span className="team-slug">{slug}</span>
+              </label>
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  },
+
+  render() {
+    let {error, loading, isInvite} = this.state;
+    return (
+      <OrganizationHomeContainer>
```

Comment:
Just a heads up we'll need to remove because of #6280

Reply:
I don't fully understand the implications here yet, looking at it again later

Reply:
nvm got it